File: src/listener.cpp
#include <ros/ros.h>
#include "std_msgs/String.h"

void sub_callback(const std_msgs::String::ConstPtr& msg) {
    ROS_INFO("I got the msg %s",msg->data.c_str()) ;
}


int main(int argc,char **argv) {
    ros::init(argc,argv,"listener") ;

    ros::NodeHandle nh ;

    ROS_INFO(" i am here") ;

    ros::Subscriber sub = nh.subscribe("chatter",10,sub_callback) ;

    ROS_INFO(" i am here") ;
 
    ros::spin() ;

    return 0 ;
}

File: scripts/listener.py
import rospy
from std_msgs.msg import String

def chat_callback(data) :
    rospy.loginfo("i have heard something",data)

def listener() :
    rospy.init_node("listener") 
    sub = rospy.Subscriber('chatter', String , callback=chat_callback)

    rospy.spin()

        
        
        

if __name__ == '__main__' :
    listener()
        
